src/commands/app/upgrade.tsx

```diff
@@ -114,6 +118,15 @@ export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
           currentApp.id,
           currentAppVersion.id,
         )) as AppAppVersion;
+    } else if (
+      this.flags["target-version"] &&
+      validateUuid(this.flags["target-version"])
+    ) {
```

Comment:
This is becoming rather long, and difficult to get your head around. Maybe extract that whole "determine target version" logic in to a function, from which you could just return early (which might also help unravel all these if/else constructs?)